### PR TITLE
PIM-9181: [Backport] PIM-9133 to 4.0 

### PR DESCRIPTION
**Pull request by anaisbaunelemaire**
**Description (for Contributor and Core Developer)**

PIM-9133 CE-master PRs => 
https://github.com/akeneo/pim-community-dev/pull/11707
https://github.com/akeneo/pim-community-dev/pull/11740

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | 
| Added legacy Behats               | 
| Added acceptance tests            | 
| Added integration tests           | 
| Changelog updated                 | Yes
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-5.0.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# 5.0.0
```
Comment (olivierpontier) [inline CHANGELOG-5.0.md:1]: The changelog file for the version 5.0 should not appear in the 4.0.x version

Comment (olivierpontier) [inline CHANGELOG-5.0.md:1]: The changelog file for the version 5.0 should not appear in the 4.0.x version

Comment (anaisbaunelemaire) [reply CHANGELOG-5.0.md:1]: Yes it seemed strange to me. I got this file by updating my 4.0 branch locally ...

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductModelController.php [hunk 2]
```diff
@@ -129,7 +113,8 @@ public function __construct(
         NormalizerInterface $violationNormalizer,
         FamilyVariantRepositoryInterface $familyVariantRepository,
         AttributeFilterInterface $productModelAttributeFilter,
-        Client $productAndProductModelClient
+        Client $productAndProductModelClient,
+        CollectionFilterInterface $productEditDataFilter
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductModelController.php:117]: BC break to handle wit h a default null value

Comment (anaisbaunelemaire) [reply src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductModelController.php]: I updated this file. Is that what you meant ?

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductModelController.php [hunk 3]
```diff
@@ -263,6 +249,7 @@ public function postAction(Request $request, int $id): Response
 
         $productModel = $this->findProductModelOr404($id);
         $data = json_decode($request->getContent(), true);
+        $data = $this->productEditDataFilter->filterCollection($data, null, ['product' => $productModel]);
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductModelController.php:252]:
```suggestion
        if (null !== $this->productEditDataFilter) {
            $data = $this->productEditDataFilter->filterCollection($data, null, ['product' => $productModel]);
        }
```

File: src/Akeneo/Pim/Enrichment/Bundle/Filter/ProductEditDataFilter.php [hunk 4]
```diff
@@ -67,16 +67,20 @@ public function supportsCollection($collection, $type, array $options = [])
         return false;
     }
 
+    private function filterData(string $type, $data)
+    {
+        if ('values' === $type) {
```
Comment (pierrejolly) [inline src/Akeneo/Pim/Enrichment/Bundle/Filter/ProductEditDataFilter.php:72]: There is no constant for this kind of data ?

**Pull request merged**